src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -21,36 +21,55 @@ internal static class CompletionHelpers
         /// <param name="wordToComplete">The word to complete.</param>
         /// <param name="possibleCompletionValues">The possible completion values to iterate.</param>
         /// <param name="toolTipMapping">The optional tool tip mapping delegate.</param>
+        /// <param name="listItemTextMapping">The optional list item text mapping delegate.</param>
         /// <param name="resultType">The optional completion result type. Default is Text.</param>
-        /// <returns></returns>
+        /// <param name="matchStrategy">The optional match strategy delegate.</param>
+        /// <returns>List of matching completion results.</returns>
         internal static IEnumerable<CompletionResult> GetMatchingResults(
             string wordToComplete,
             IEnumerable<string> possibleCompletionValues,
             Func<string, string> toolTipMapping = null,
-            CompletionResultType resultType = CompletionResultType.Text)
+            Func<string, string> listItemTextMapping = null,
+            CompletionResultType resultType = CompletionResultType.Text,
+            MatchStrategy matchStrategy = null)
```

Comment:
Maybe assign default here?
```suggestion
            MatchStrategy matchStrategy = DefaultMatch)
```

Reply:
Yeah this doesn't work, can't assign delegate default since it must be compile time constant.